tests/spec/core/linter-rules/check-punctuation.js

```diff
@@ -0,0 +1,61 @@
+fdescribe("Core Linter Rule - 'check-punctuation'", () => {
+  const ruleName = "check-punctuation";
+  const config = { lint: { [ruleName]: true } };
+  let rule;
+  beforeAll(async () => {
+    rule = await new Promise(resolve => {
+      require([`core/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
+    });
+  });
+  it("checks p ending without a punctuation", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
+          <p>Should fail.</p>
+          <p id="no-fullstop-at-end">Should. pass</p>
+          <p>Should. Fail.</p>
+          <p id="unnecessary-tests-after-fullstop">Should Pass. Tests</p>
+        </section>
+    `;
+
+    const noFullStop = doc.getElementById("no-fullstop");
+    const noFullStopAtEnd = doc.getElementById("no-fullstop-at-end");
+    // const unnecessaryTestsAfterFullStop = doc.getElementById("unnecessary-tests-after-fullstop");
+    
+    const results = await rule.lint(config, doc);
+    const [result] = results;
+  
+    expect(result.offendingElements.length).toEqual(2);
+    expect(result.offendingElements[0]).toEqual(noFullStop);
+    expect(result.offendingElements[1]).toEqual(noFullStopAtEnd);
+  });
+  it("checks error message for p ending without a punctuation", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
+          <p>Should fail.</p>
+          <p id="no-fullstop-at-end">Should. pass</p>
+          <p>Should. Fail.</p>
+          <p id="unnecessary-tests-after-fullstop">Should Pass. Tests</p>
+        </section>
+    `;
+  
+    const results = await rule.lint(config, doc);
+    const [result] = results;
+    
+    const noFullStop = doc.getElementById("no-fullstop");
+    const noFullStopAtEnd = doc.getElementById("no-fullstop-at-end");
+    const unnecessaryTestsAfterFullStop = doc.getElementById("unnecessary-tests-after-fullstop");
+    
+    expect(result.name).toEqual(ruleName);
+    expect(result.occurrences).toEqual(2);
+    expect(result.description).toEqual("`<p>` tags should end with either of . | : | ! | ? ");
```

Comment:
Here, we should just expect that the description `.toBeTruthy()`, as all we really care about is that some description is given. Same with `howToFix`.

Reply:
Done :+1: